src/abstractdataframe/abstractdataframe.jl

```diff
@@ -970,15 +970,52 @@ julia> dropmissing(df, [:x, :y])
 @inline function dropmissing(df::AbstractDataFrame,
                              cols::Union{ColumnIndex, MultiColumnIndex}=:;
                              view::Bool=false, disallowmissing::Bool=!view)
+    # Identify Bool mask of which rows have no Missings
     rowidxs = completecases(df, cols)
     if view
         if disallowmissing
             throw(ArgumentError("disallowmissing=true is incompatible with view=true"))
         end
         return Base.view(df, rowidxs, :)
     else
-        newdf = df[rowidxs, :]
-        disallowmissing && disallowmissing!(newdf, cols)
+        # With exact indices we can skip a lot of iterations (as opposed to indexing via a mask)
+        selected_rows = _findall(rowidxs)
+        new_columns = Vector{AbstractVector}(undef, ncol(df))
+        df_columns = if typeof(df) == DataFrame 
+            _columns(df)
+        else 
+            # for SubDataFrame
+            collect(eachcol(df))
+        end
+
+        # What column indices should disallowmissing be applied to
+        cols_inds = index(df)[cols]
+
+        # Threading decision rule borrowed from `_threaded_getindex`
+        if Threads.nthreads() > 1 && ncol(df) > 1 && length(selected_rows) >= 1_000_000
+            @sync for i in eachindex(new_columns)
+                # for each column, check if disallowmissing should be applied
+                Threads.@spawn if disallowmissing && (i in cols_inds)
+                    new_columns[i] = Missings.disallowmissing(@view df_columns[i][selected_rows])
+                else
+                    new_columns[i] = df_columns[i][selected_rows]
+                end
+            end
+        else
+            for i in eachindex(new_columns)
+                if disallowmissing && (i in cols_inds)
+                    new_columns[i] = Missings.disallowmissing(@view df_columns[i][selected_rows])
```

Comment:
```suggestion
                    new_columns[i] = disallowmissing(view(df_columns[i], selected_rows))
```

Reply:
Makes sense with one change -- I need to keep the explicit import from Missings.
The `Missings.disallowmissing` function clashes with the local scope keyword `disallowmissing`.

The alternatively would be to change the keyword name, but that change the user-facing API, which is completely unnecessary.

Reply:
Could you check if the function works for you by changing `@view` to `view`?

When I run it with `@view`, all is good.
When I use `view()`, I get an error, as if `disallowmissing` variable and function name were clashing:
(this is the exact error you'd get if you remove the explicit import of function `disallowmissing()`, because the function scope will override it)

> ERROR: MethodError: objects of type Bool are not callable
Maybe you forgot to use an operator such as *, ^, %, / etc. ?
Stacktrace:
  [1] #dropmissing#1056
    @ ~/Documents/GitHub/DataFrames.jl/src/abstractdataframe/abstractdataframe.jl:1005 [inlined]
  [2] dropmissing (repeats 2 times)
    @ ~/Documents/GitHub/DataFrames.jl/src/abstractdataframe/abstractdataframe.jl:970 [inlined]

It SHOULD work - I don't see why not and it works fine in smaller snippets I created trying to debug it / reproduce it (see f1 vs f2 below). 
The only context it fails in is within the `dropmissing` function.

_Can you make this change on your machine and run `dropmissing(df)` ?_


Failed attempts to reproduce the bug. 
LLVM is identical, so I wonder if it's something on the Julia parsing level (issue with references references)
```
function f1(df,selected_rows;disallowmissing=true)
    Missings.disallowmissing(@view(DataFrames._columns(df)[1][selected_rows]))
end
function f2(df,selected_rows;disallowmissing=true)
    Missings.disallowmissing(view(DataFrames._columns(df)[1],selected_rows))
end
function f3(df,selected_rows;disallowmissing=true)
    new_columns = Vector{AbstractVector}(undef, ncol(df))
    df_columns = df isa DataFrame ? DataFrames._columns(df) : collect(AbstractVector, eachcol(df))
    for i in eachindex(df_columns)
        if disallowmissing
            new_columns[i] = Missings.disallowmissing(view(df_columns[i],selected_rows))
        end
    end
    new_columns
end
basecase = (; rows_total=10^3, cols_total=10, cols_missing=3, missingness_share=0.1)
df = generate_data(; basecase...)
rowidxs = completecases(df)
selected_rows = DataFrames.findall(rowidxs)

# no failures -- all runs fine
f1(df,selected_rows)
f2(df,selected_rows)
f3(df,selected_rows)
dropmissing(df)

@code_lowered f1(df,selected_rows)
@code_lowered f2(df,selected_rows)
@code_llvm f1(df,selected_rows)
@code_llvm f2(df,selected_rows)
```

Reply:
Oh, I'm slow. It was the same issue but with the other keyword -- we use keyword `view`, which overwrites the call to `view()`

I'll import it implicitly